### ISSUE TITLE
ec2 server create fails with error ERROR: undefined local variable or method `i' for #<Chef::Knife::Ec2ServerCreate:0x00000000090946e8> for option --iam-profile

### ISSUE THREAD

**Issue by dheerajd-msys** [chef/knife-ec2#591]
# Version:

knife-ec2 v1.0.0 and chef v15.1.8 on workstation

# Environment:

Doesn't matter

# Scenario:

```
bundle exec knife ec2 server create -I ami-005bdb005fb00e791 -N dh-ubu-14  -U ubuntu --connection-protocol ssh --flavor t2.micro -c E:\Backup\Project\chef-starter\chef-repo\.chef\knife.rb -VV --sudo --yes --iam-profile dh-role
```
# Steps to Reproduce:

Please run above command to create the server from workstation.

# Expected Result:

Server should get created and bootstrapped.

# Actual Result:

```
ERROR: undefined local variable or method `i' for #<Chef::Knife::Ec2ServerCreate:0x00000000090946e8>
```
Fix would be at https://github.com/chef/knife-ec2/pull/587/files#diff-0da44873959aafda9bacd02cd08c8f5cR142

**Comment by dheerajd-msys** [comment]
@vsingh-msys Please fix this in https://github.com/chef/knife-ec2/pull/587

**Comment by vsingh-msys** [comment]
Fixed in commit 2a8adc8 of #587
